pilot/pkg/model/service.go

```diff
@@ -70,11 +79,11 @@ type Service struct {
 	// CreationTime records the time this service was created, if available.
 	CreationTime time.Time `json:"creationTime,omitempty"`
 
-	// Name of the service, e.g. "catalog.mystore.com"
-	Hostname host.Name `json:"hostname"`
```

Comment:
Please revert, it is so widely used.  If you want to support mcs, it is better to add a new field

Reply:
I don't think the fact that it's used widely is a sufficient reason to keep it. If we're introducing multiple hostnames, leaving this creates confusion and room for misuse.

I'd like to get thoughts from others: @howardjohn @jacob-delgado thoughts?

Reply:
One thing we could do is:
```
	// ClusterLocal specifies the cluster.local host and cluster VIPs for this service.
	HostVIPs `json:"clusterLocalHostVIPs,omitempty"`
```

Then access can just be like `svc.Hostname`. When we add clusterset, it would be `ClusterSet HostVIPs` so access is like `svc.ClusterSet.Hostname`. The benefit here is its simpler and minimal changes; the downside is its less explicit.

So I guess if we want all callers to ask themselves: "Should I use the ClusterLocal or the ClusterSet IP?", then they way we have now is probably right. If instead using ClusterLocal is right in most cases and we only need the ClusterSet one in some isolated multicluster code, then I think embedded or other approach may be simpler.

I think right now we are in the latter case (using ClusterLocal is right in most cases - really all cases today as ClusterSet does not exist). Do we expect that to change in the near future?

Reply:
> So I guess if we want all callers to ask themselves: "Should I use the ClusterLocal or the ClusterSet IP?", then they way we have now is probably right.

This is exactly my thinking. By making each call site use `svc.ClusterLocal.xxx`, we give the fields themselves way more context, so it will be easier to identify bugs going forward.

> I think right now we are in the latter case (using ClusterLocal is right in most cases - really all cases today as ClusterSet does not exist). Do we expect that to change in the near future?

I do expect this to change in the future. I expect us to get to a point where MCS is a default feature of k8s and Istio uses it by default to drive service discoverability. IMO it would be better to just take the refactoring hit now, and remove all of the assumptions of a logical default.

Reply:
I point it because, model.Service is a core API used internally.  We are in Day-2 stage As i know, how would people think?


Another reason is MCS is still an alpha feature, not sure how many users. It is not wise to introduce an alpha support but may break a stable one.

Reply:
`model.Service` is only used internally. Adding it to be cluster-aware will help devs know the specific places where MCS is different. We've had issues trying to modifying `model.Service` by just adding new, similar fields, but it has made finding subtle issues hard to track. I think this is the right approach.

Reply:
I prefer @howardjohn 's suggestion inlined field makes minor change to istio, but since you have manually changed all places, i am ok with either.